### PR TITLE
Fix cortex_ingester_tsdb_head_max_time_seconds

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR fixes `cortex_ingester_tsdb_head_max_time_seconds` metric introduced in https://github.com/grafana/mimir/pull/5786.

`cortex_ingester_tsdb_head_max_time_seconds` is based on `prometheus_tsdb_head_max_time_seconds`, which however is NOT exported by TSDB library. TSDB library only exports `prometheus_tsdb_head_max_time` (without `_seconds` suffix). Metric with `_seconds` suffix is exported by Prometheus `main` package here:

https://github.com/grafana/mimir-prometheus/blob/32d87282addfaa2af9f7a6b953d17f3da6101d66/cmd/prometheus/main.go#L1194-L1201

~We fix Mimir metric by exporting the `prometheus_tsdb_head_max_time_seconds` on our own, registered into per-TSDB registry. Note that Prometheus metric exports it with seconds precision because of integer division, in Mimir we do float division instead.~

**Update:** We now compute max timestamp in the ingester, which simplifies the logic. Metric name was changed to `cortex_ingester_tsdb_head_max_timestamp_seconds` and "min" version was also added (`cortex_ingester_tsdb_head_min_timestamp_seconds`)

#### Checklist

- [x] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -2151,6 +2151,13 @@ func (i *Ingester) createTSDB(userID string, walReplayConcurrency int) (*userTSD
 	}
 	db.DisableCompactions() // we will compact on our own schedule
 
+	// Register metric that reports Head MaxTime in seconds.
+	// (TSDB library already exports prometheus_tsdb_head_max_time metric, but it's unit-neutral.)
+	promauto.With(tsdbPromReg).NewGaugeFunc(prometheus.GaugeOpts{
```
Comment (pracucci) [inline pkg/ingester/ingester.go:2156]: Now it looks quite a complicated wiring. Have you considered just defining the GaugeFunc in the main registry (not the per-TSDB one), iterate through TSDBs and get the max head time value?

Comment (pstibrany) [reply pkg/ingester/ingester.go:2156]: I don't see this very complicated, but your suggestion is certainly a possibility as well, and will be simpler. Let me do that.

Comment (pstibrany) [reply pkg/ingester/ingester.go]: I've implemented this change in https://github.com/grafana/mimir/pull/5815/commits/3b8c2a79c72f35b04aef73a75f06b4c4a31e1728.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**